GoogleForCreators/web-stories-wp
# New keyboard shortcut dialog

## Context

This completely revamps the keyboard shortcut menu to the new design and design system. See todo below and parent ticket for more details.

## To-do

- [x] Update dialog content as per design
- [x] Update dialog placement and background as per design
- [x] Make sure dialog scrolls correctly on a smaller viewport
- [x] Make sure functionality is covered by visual tests

## User-facing changes

_When open:_
| Before | After |
|-|-|
| ![Screen Shot 2021-02-16 at 01 02 52](https://user-images.githubusercontent.com/637548/108024639-f2ba5a80-6ff2-11eb-92bc-add4b8b15ad6.png) | ![Screen Shot 2021-02-16 at 01 03 34](https://user-images.githubusercontent.com/637548/108024634-f0f09700-6ff2-11eb-8af4-117eedb47b6b.png) |

_Sizing (height):_
| With scroll | Without scroll | With meta-boxes |
|-|-|-|
| ![Screen Shot 2021-02-16 at 01 03 34](https://user-images.githubusercontent.com/637548/108024634-f0f09700-6ff2-11eb-8af4-117eedb47b6b.png) | ![Screen Shot 2021-02-16 at 01 11 45](https://user-images.githubusercontent.com/637548/108025169-00241480-6ff4-11eb-9f19-0530ae7a736f.png) | ![Screen Shot 2021-02-16 at 01 11 31](https://user-images.githubusercontent.com/637548/108025147-f6021600-6ff3-11eb-8665-dcd8ee5d638f.png) |

_Opening and closing:_
| By mouse | By keyboard | In RTL |
|-|-|-|
| ![kb-mouse](https://user-images.githubusercontent.com/637548/108024977-9efc4100-6ff3-11eb-8fd8-8babe00a0d87.gif) | ![kb-keyboard](https://user-images.githubusercontent.com/637548/108024990-a7547c00-6ff3-11eb-86aa-144b88c8301f.gif) | ![kb-rtl-proper](https://user-images.githubusercontent.com/637548/108027172-b806f100-6ff7-11eb-89d8-1396a39bde03.gif) |

## Testing Instructions

### QA

This PR can be tested by following these steps:

1. Open the keyboard shortcut menu by either mouse or keyboard
2. Observe that the menu looks good
3. Press <kbd>esc</kbd>, or press <kbd>space</kbd> or <kbd>enter</kbd> with the close button focused, or click outside the dialog to close it
4. Also observe, that the content is sized correctly in different browser heights (and with and without meta-boxes open)

### UAT




## Reviews

## Checklist




---

Fixes #6385
